app/screens/ScheduleScreen/ScheduleScreen.tsx

```diff
@@ -11,63 +11,115 @@ import ScheduleCard, { ScheduleCardProps, Variants } from "./ScheduleCard"
 import { useStores } from "../../models"
 import { formatDate } from "../../utils/formatDate"
 import { useAppNavigation } from "../../hooks"
+import Animated, {
+  useAnimatedScrollHandler,
+  useSharedValue,
+  runOnJS,
+} from "react-native-reanimated"
+
+const { width } = Dimensions.get("window")
 
 export const ScheduleScreen: FC<TabScreenProps<"Schedule">> = observer(function ScheduleScreen() {
   useHeader({ title: "Schedule" })
   const navigation = useAppNavigation()
   const { schedulesStore } = useStores()
-  const { fetchData, selectedSchedule } = schedulesStore
+  const { fetchData, setSelectedSchedule, selectedSchedule, schedules } = schedulesStore
 
   useLayoutEffect(() => {
     fetchData()
   }, [fetchData])
 
+  const updateSchedule = (index) => setSelectedSchedule(schedules[index])
+
+  const scrollX = useSharedValue(0)
+  const ref = React.useRef(null)
+  const scrollHandler = useAnimatedScrollHandler({
+    onScroll: (event) => {
+      scrollX.value = event.contentOffset.x
+    },
+    onMomentumEnd: (event) => {
+      const contentOffset = event.contentOffset
+      const viewSize = event.layoutMeasurement
+
+      // Divide the horizontal offset by the width of the view to see which page is visible
+      const index = Math.floor(contentOffset.x / viewSize.width)
+
+      // ? Why does this work this way with MST?
```

Comment:
Update the store when the scrolling stops with what page we landed on.

I'm not sure why I ran into an MST error when including `schedules` in here directly